### PR TITLE
add column checked on state.block

### LINKED ISSUE #3541: synchronizer: each block on L1 stored on DB must have a bool that says if the block need to be checked or not
# Rationale

When insert a block on node DB must mark:
- If, in the moment of the request, the block_number is over **finalized** block need to be checked
- If a second process verifiy the hash of this block when the block_number is below L1 **finalized**  block mark it

# Implementation

Add a column called **checked**:
- If is created below **L1 finalized** mark as **true** because don't need to be verified

### PR DESCRIPTION
**Pull request by joanestebanr**
Closes #3541
### What does this PR do?

add column **checked** to state.block

### Reviewers

Main reviewers:

- @tclemos 
- @ARR552 
- @agnusmor 

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: db/migrations/state/0018.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- +migrate Up
+ALTER TABLE state.block
+    ADD COLUMN IF NOT EXISTS checked BOOL NOT NULL DEFAULT TRUE;
```
Comment (ARR552) [inline db/migrations/state/0018.sql]: In a full resync it is perfect but in the current env, If we do so, maybe we miss some previous mismatch. becuase we were assuming tha it was ok. Wdyt?

File: db/migrations/state/0018.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+-- +migrate Down
+ALTER TABLE state.receipt
+    DROP COLUMN IF EXISTS checked;
```
Comment (ARR552) [inline db/migrations/state/0018.sql]: Can you give more context about this change?

**Review by ARR552 — APPROVED** [review]

File: db/migrations/state/0018.sql (hunk 2 above)
Comment (joanestebanr) [reply db/migrations/state/0018.sql]: my bad, wrong table!

File: db/migrations/state/0018.sql (hunk 1 above)
Comment (joanestebanr) [reply db/migrations/state/0018.sql]: I agree, I changed and the latest 1000 blocks are checked=false

**Pull request merged**